docs/theme.md

```diff
@@ -0,0 +1,138 @@
+
+To use **theme components** inside your own component definitions, your component must pull them from the theme object within contexture-react's `ThemeContext`.
+
+There are two ways to get the theme from context: the `ThemeConsumer` component, and the `withTheme` function.
+
+### ThemeConsumer
+
+The `ThemeConsumer` helper component takes a render function as its child, and passes the **theme object** from context into that function. This example demonstrates how to create a basic IconButton component using the `Icon` and `Button` **theme components**, plus `ThemeConsumer`:
+
+```jsx
+// IconButton.js
+
+import React from 'react'
+import { ThemeConsumer } from 'contexture-react/src/utils/theme'
+
+let IconButton = ({ icon, children, ...props }) => (
+  <ThemeConsumer>
+    {theme =>
+      <theme.Button {...props}>
+        <theme.Icon icon={icon} />
+        {children}
+      </theme.Button>
+    }
+  </ThemeConsumer>
+)
+export default IconButton
+```
+
+### withTheme
+
+The `withTheme` higher-order component injects the theme object from context into its component argument as the `theme` prop. Here is the same IconButton example using `withTheme`:
+
+```jsx
+// IconButton.js
+
+import React from 'react'
+import { withTheme } from 'contexture-react/src/utils/theme'
+
+let IconButton = ({ theme, icon, children, ...props }) => (
+  <theme.Button {...props}>
+    <theme.Icon icon={icon} />
+    {children}
+  </theme.Button>
+)
+export default withTheme(IconButton)
+```
+
+`ThemeConsumer` and `withTheme` both accomplish the same thing in different ways. You're free to use whichever one you like best. 🙂
+
+
+## Default theme components
+
+The default theme in contexture-react is GreyVest. The full list of **theme components** included in GreyVest is shown in the table below.
+
+| Key | Expected props | Notes |
+| --- | --- | --- |
+| `AdderPicker` | `options`, `onChange`, `label` | *this key is actually based on specific usage in the FilterAdder component, and should probably just be removed from the defaults in favor of a nested style* |
+| `BarChart` | `height`, `borderColor`, `min`, `max` | Used in the DateHistogram and TermsStats example types |
+| `Box` | `children` | A generic container element |
+| `Button` | `isActive`, `primary`, `onClick`, `children` | A generic button |
+| `Checkbox` | `checked`, `onChange` | A generic checkbox |
+| `CheckButton` | `checked`, `onClick`, `children` | A button with an `onClick` handler and an on/off state, containing a checkbox that displays this state *(this really should be extrapolated to some kind of generic BinaryStateButton)* |
+| `ExpandableTable` | `data`, `columns`, `recordKey`, `sortField`, `sortDir`, `columnSort` | Used in the TermsStatsTable and CheckableTermsStatsTable example types |
+| `Icon` | `icon`, `onClick` | A generic icon component |
+| `Fonts` | none | For holding `<link>` elements to fonts used in the theme |
+| `Input` | same as basic `input` | A generic input component |
```

Comment:
Does anything actually use this directly? With example types, we started to have explicit NumberInput, TextInput, etc.

There are also definitely a bunch of missing things like `Select`, `Radio`, etc

Reply:
Yep, but it's really the same as TextInput - best to keep it consistent with the others, so Input should get renamed.

Select was definitely needed. I don't believe anything uses a Radio component though - pretty sure it's just RadioList